src/main/java/com/iota/iri/utils/dag/DAGUtils.java

```diff
@@ -0,0 +1,366 @@
+package com.iota.iri.utils.dag;
+
+import com.iota.iri.controllers.MilestoneViewModel;
+import com.iota.iri.controllers.TransactionViewModel;
+import com.iota.iri.model.Hash;
+import com.iota.iri.storage.Tangle;
+
+import java.util.*;
+
+/**
+ * This class offers generic functions for recurring tasks that are related to the tangle and that otherwise would have
+ * to be implemented over and over again in different parts of the code.
+ */
+public class DAGUtils {
+    /**
+     * Holds references to the singleton DAGUtils instances.
+     */
+    protected static HashMap<Object, DAGUtils> instances = new HashMap<>();
```

Comment:
It doesn't make a difference and i was thinking about adding additional parameters at some point but i can change that so its conistent for now

Reply:
It is for consistency and readability.

Reply:
1. I agree with @karimodm, please change to `Tangle`. `Object` should be used as last resort imo. It gives lots of flexibility on one hand, but as the code grows too much flexibility can cause confusing design which leads to waste of time and bugs.
2. Since this is a static object which we only initialize in the constructor I would also add the `final` word to it. I don't remember if I ranted to you how much I usually hate the overuse of `final` in the code, but here it should be written down because: 

a. It is a general convention when creating static fields in java
b. It ensures you don't change the reference later on (you are dealing with global scope here)
c. It provides gurantees about references not escaping in multithreaded environments.(https://wiki.sei.cmu.edu/confluence/display/java/OBJ10-J.+Do+not+use+public+static+nonfinal+fields).

Now you may claim that the reasons I enlisted here may less affect the current code. However I think that sticking to conventions usually help preventing future bugs created by other programmers that come after you. I also need subconcientiously see it.


I will also rant about `protected`. Unless you need this inherited I will keep it as private. Here I don't have much good arguements about why not just make everything protected so inherited class will always have access. I just think it a good principle to expose only what you need.